more_itertools/more.py

```diff
@@ -3900,3 +3901,71 @@ def __next__(self):
         self.items_seen += 1
 
         return item
+
+
+def chunked_even(iterable, n):
+    """Break *iterable* into lists of length *n*, but in contrast with `chunked`,
+        items are distributed as evenly as possible so that all lists' lengths
+        are within 1 of each other:
+
+            >>> list(chunked_even([1, 2, 3, 4, 5, 6, 7], 3))
+            [[1, 2, 3], [4, 5], [6, 7]]
+
+        If *iterable* has `__len__`, O(n) space is used, with O(n) time per list.
+        Otherwise, O(n^2) space is needed, and O(n^2) time per list.
+    """
+    
+    len_method = getattr(iterable, '__len__', None)
+    
+    if len_method is None:
+        return _chunked_even_online(iterable, n)
+    else:
+        return _chunked_even_finite(iterable, len_method(), n)
+
+
+def _chunked_even_online(iterable, n):
+    buffer = []
+    # TODO: n*(n - 1) - 2 is not enough (tests fail), but this might be. Proof?
+    maxbuf = n*(n - 1) - 1
+    for x in iterable:
+        buffer.append(x)
+        if len(buffer) == maxbuf:
+            yield buffer[:n]
+            buffer = buffer[n:]
+    yield from _chunked_even_finite(buffer, len(buffer), n)
+
+
+def _chunked_even_finite(iterable, N, n):
+    if N < 1:
+        return
+    
+    # Every list is either size `full_size <= n` or `partial_size = full_size - 1`
+    num_lists = ceil(N / n)
```

Comment:
Could you convert this to integer math? I'd rather not import the `math` module.